### PR TITLE
Implement devise

### PR DESCRIPTION
**Pull request by mgmediaweb**
## Features
- Install the gem Devise.
- Clear the database and run a new migration.
- Make necessary changes in **routes** and **controllers**.
- Implement the design into the login views.
- All Rubocop and Testing passed.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 class PostsController < ApplicationController
+  # before_action :authenticate_user!   -----this line should be added as well on the users controller
   def index
     @user = User.find(params[:user_id])
     # @posts = Post.where(author: @user).includes([:author])
```
Comment (rustam-tolipov) [inline app/controllers/posts_controller.rb:6]: - [ ] Kindly remove commented code from your file.

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
 class PostsController < ApplicationController
+  # before_action :authenticate_user!   -----this line should be added as well on the users controller
```
Comment (rustam-tolipov) [inline app/controllers/posts_controller.rb]: [OPTIONAL] You can add this line for the controller you want or you can add it application_controller to share the same methods all at once.

File: Gemfile [hunk 3]
```diff
@@ -3,6 +3,8 @@ git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
 ruby '3.1.2'
 
+gem 'devise'
```
Comment (rustam-tolipov) [inline Gemfile:6]: [OPTIONAL] Kindly personalize your MIT license to your project.

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (mgmediaweb) [reply app/controllers/posts_controller.rb:6]: ooh sorry for this mistake, now its solved

**Review by mateo951 — APPROVED** [review]
### Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🐐 
### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**